wordpress-mobile/WordPress-Android
# Login rework: Site address input screen

This PR introduces the site address input screen. This screen is used as the first step for logging in a self-hosted site.

At the moment, the screen doesn't implement the case of a successful site address discovery. Also, the final visual touches, the messages and final texts will be done in a future PR.

To test:
* With the app data cleaned up, launch the app
* Tap on the "Login" button to go to the email screen
* Tap on the "Log in to your site by entering your site address instead." link on the lower left to go to the site address input screen
* Tap on the "Need help finding your site address?" link at the lower left
* Notice the dialog popping up showing a small help about how to find the site address. On this dialog, the "NEED MORE HELP?" button is not implemented yet. Tap "OK" to return to the site address screen
* Try to log into any site (self-hosted or wpcom) and observe the potential error or success
* Try to log into http://do.wpmt.co/httpauth/. An HTTP Auth dialog will pop up asking for the username/password pair for HTTP Basic Auth.
* Try to log into https://do2.wpmt.co/vanilla/. A confirmation dialog should pop up asking for a confirmation of the self-signed SSL certificate.
